File: frontend-python/models/player.py
class PlayerFormData:
    def __init__(self, form):
        self.fullName = form.get("fullName")
        self.age = int(form.get("age") or 0)
        self.role = form.get("role")
        self.battingStyle = form.get("battingStyle")
        self.bowlingStyle = form.get("bowlingStyle")
        self.batchId = form.get("batchId") or None

    def to_dict(self):
        return {
            "fullName": self.fullName,
            "age": self.age,
            "role": self.role,
            "battingStyle": self.battingStyle,
            "bowlingStyle": self.bowlingStyle,
            "batch": None if self.batchId is None else {"id": int(self.batchId)},
        }

File: backend-springboot/src/main/java/com/example/cricketacademy/entity/Player.java
package com.example.cricketacademy.entity;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "players")
public class Player {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String fullName;

    private Integer age;

    @Column(nullable = false)
    private String role;

    private String battingStyle;
    private String bowlingStyle;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "batch_id")
    private Batch batch;

    private LocalDate joinDate;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getFullName() { return fullName; }
    public void setFullName(String fullName) { this.fullName = fullName; }

    public Integer getAge() { return age; }
    public void setAge(Integer age) { this.age = age; }

    public String getRole() { return role; }
    public void setRole(String role) { this.role = role; }

    public String getBattingStyle() { return battingStyle; }
    public void setBattingStyle(String battingStyle) { this.battingStyle = battingStyle; }

    public String getBowlingStyle() { return bowlingStyle; }
    public void setBowlingStyle(String bowlingStyle) { this.bowlingStyle = bowlingStyle; }

    public Batch getBatch() { return batch; }
    public void setBatch(Batch batch) { this.batch = batch; }

    public LocalDate getJoinDate() { return joinDate; }
    public void setJoinDate(LocalDate joinDate) { this.joinDate = joinDate; }
}
